### ISSUE TITLE
Imbalanced GPU memory

### ISSUE THREAD

**Issue by MJITG** [DeepMotionAIResearch/DenseMatchingBenchmark#10]
When using acfnet code, the program takes a lot of memory in GPU0 but much less memory in others. Why would this happen?

**Comment by youmi-zym** [comment]
Hi @MJITG ,
I think the problems happened mainly because some operations (e.g., loss computation) take only on GPU0 but not parallel/distribute to other GPUS.  
If so, 
1. [torch.distributed](https://pytorch.org/docs/stable/distributed.html) is recommended, and our architecture is based on distributed mode and the consumed memory on each GPU is the same, so you can try it.
2. you can combine the model and loss as a wholistic nn.Module (e.g., define a module: Class ModelAndLoss) and then use either torch.nn.Parallel or torch.distruted to distribute your model to each GPU
I hope I got your problem.

**Comment by MJITG** [comment]
Thanks for your advice! I'll try later
